### PR TITLE
Test pj specs pubsub config.

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Testing prowjobs definitions to make sure they have expected and required settings. Test are written using ginkgo framework as it gives nice way of writing descriptive, human readable tests in a code itself. There is no need to provide additional file mapping a test code to the human friendly description.

Changes proposed in this pull request:

- Added ginkgo tests checking pubsubs config in prowjobs.
- Added prowjobs testing against all repos from list.
- Added building ginkgo image to run tests.
- Dependency are maintained by dependabot.

### CODE REVIEW

**Comment by dekiel** [comment]
/hold
Hold merging PR till failing tests for prowjobs will be fixed as this will stop other changes.

**Comment by dekiel** [comment]
Waiting for #6865 to be merged first.

**Comment by halamix2** [comment]
`pull-unit-test-go`is failing

**Comment by dekiel** [comment]
@halamix2 `pull-unit-test-go` was failing because not merged fixes from #6865. Once fixes were merged a test is green. Please review.

File: prow/images/ginkgo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.19.4-alpine3.17 AS builder
```
Comment (halamix2) [inline prow/images/ginkgo/Dockerfile:1]: There is golang:1.20.0-alpine3.17 already, but dependabot will bump it anyway

**Pull request merged**